packages/rainbowkit/src/components/ProfileDetails/ProfileDetails.tsx

```diff
@@ -32,10 +33,21 @@ export function ProfileDetails({
   ensName,
   onClose,
   onDisconnect,
+  setBalanceData,
 }: ProfileDetailsProps) {
   const showRecentTransactions = useContext(ShowRecentTransactionsContext);
   const [copiedAddress, setCopiedAddress] = useState(false);
 
+  const { data: wagmiBalanceData } = useBalance({
+    address,
+    enabled: !balanceData,
+  });
+
+  // biome-ignore lint/nursery/useExhaustiveDependencies: <explanation>
+  useEffect(() => {
+    if (wagmiBalanceData) setBalanceData(wagmiBalanceData);
+  }, [wagmiBalanceData?.value]);
+
```

Comment:
Does the `balanceData` prop that's passed down here trigger the same behavior upon re-render?